GTG/backends/generic_backend.py

```diff
@@ -276,12 +276,14 @@ def __init__(self, parameters):
             # if it's not specified, then this is the default backend
             # (for retro-compatibility with the GTG 0.2 series)
             parameters[self.KEY_DEFAULT_BACKEND] = True
+
         # default backends should get all the tasks
-        if parameters[self.KEY_DEFAULT_BACKEND] or \
-                (self.KEY_ATTACHED_TAGS not in parameters and
-                 self._general_description[self.BACKEND_TYPE] ==
-                    self.TYPE_READWRITE):
+        attach_tags = self.KEY_ATTACHED_TAGS not in parameters and \
+            self._general_description[self.BACKEND_TYPE] == self.TYPE_READWRITE
+
+        if parameters[self.KEY_DEFAULT_BACKEND] or attach_tags:
             parameters[self.KEY_ATTACHED_TAGS] = [ALLTASKS_TAG]
```

Comment:
I see that you've separated the conditions, but `attach_tags` also contains whenever the backend type is read and writable. Also, it says "not in parameters", so `no_attached_tags`?

Reply:
Makes sense, will update it